### PR TITLE
Paint background color of root element over the entire canvas

### LINKED ISSUE #19894: Background colour is drawn over elements with negative z-index
```html
<style>
html {
  background-color: #7D2663;
}

h1 {
  position: absolute;
  z-index: -1;
}
</style>
<h1>Web</h1>
```

Firefox:
screen shot 2018-01-29 at 8 03 00 am

Servo:
screen shot 2018-01-29 at 8 03 19 am

### PR DESCRIPTION
**Pull request by ferjm**
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #19894

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20057)

### CODE REVIEW

File: components/layout/display_list/builder.rs [hunk 1]
```diff
@@ -463,7 +463,8 @@ impl<'a> DisplayListBuildState<'a> {
         // refer to the steps in CSS 2.1 Appendix E.
         // Steps 1 and 2: Borders and background for the root.
         while child_items.last().map_or(false, |child| {
-            child.section() == DisplayListSection::BackgroundAndBorders
+            (child.section() == DisplayListSection::BackgroundAndBorders) |
```
Comment (emilio) [inline components/layout/display_list/builder.rs]: nit: use `matches!`, or `||`?

**Comment by pyfisch** [comment]
Looks good.

r+

(Probably someone else needs to approve this for bors)

**Comment by jdm** [comment]
@bors-servo r-
What about an automated test?

**Comment by jdm** [comment]
Closing due to inactivity.

**Pull request closed without merging**